### PR TITLE
refactor: override HasStyle implementation in Notification

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

Same as #4549 and #4576 but for `Notification`.

Based on #4573

## Type of change

- Refactor

### CODE REVIEW

File: vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java [hunk 1]
```diff
@@ -598,9 +598,25 @@ protected void onDetach(DetachEvent detachEvent) {
         });
     }
 
-    private void initConnector() {
-        getElement().executeJs(
-                "window.Vaadin.Flow.notificationConnector.initLazy(this)");
+    /**
+     * Sets the CSS class names of the dialog overlay element. This method
```
Comment (sissbruecker) [inline vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java:602]:
```suggestion
     * Sets the CSS class names of the notification overlay element. This method
```

File: vaadin-notification-flow-parent/vaadin-notification-flow/src/test/java/com/vaadin/flow/component/notification/NotificationHasStyleTest.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+package com.vaadin.flow.component.notification;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.component.html.Div;
```
Comment (sissbruecker) [inline vaadin-notification-flow-parent/vaadin-notification-flow/src/test/java/com/vaadin/flow/component/notification/NotificationHasStyleTest.java:25]:
```suggestion
import com.vaadin.flow.component.html.Div;
```
Unused import.

File: vaadin-notification-flow-parent/vaadin-notification-flow/pom.xml [hunk 3]
```diff
@@ -39,6 +39,11 @@
             <artifactId>flow-test-util</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>com.vaadin</groupId>
+            <artifactId>flow-html-components</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (sissbruecker) [inline vaadin-notification-flow-parent/vaadin-notification-flow/pom.xml:46]: Seems this is not needed anymore after removing the unused import.

**Review by sissbruecker — APPROVED** [review]

File: vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java (hunk 1 above)
Comment (web-padawan) [reply vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java]: Fixed.

File: vaadin-notification-flow-parent/vaadin-notification-flow/pom.xml (hunk 3 above)
Comment (web-padawan) [reply vaadin-notification-flow-parent/vaadin-notification-flow/pom.xml]: Thanks, removed.

**Pull request merged**